packages/gatsby-plugin-fastify/src/utils/headers.ts

```diff
@@ -38,3 +47,34 @@ export function moduleHeaderDecorator(this: FastifyReply, module: Modules): void
 export function setHeaderDecorator(this: FastifyReply, key: string, value: string) {
   this.header(key, value);
 }
+
+export function appendRouteHeaders(route: string, reply: FastifyReply): Record<string, string> {
+  const {
+    server: { headers },
+  } = getConfig();
+
+  /**
+   * TODO: Solve This
+   * If the route is a match for a header, return the header value
+   * But how do we prioritize certain matches.
+   * How do I make sure that the header is not overwritten by other matches?
+   * we expect this but we need a way to prioritize one match over another
+   * or am I over complicating this?
+   *
+   * @tsdexter: I think we can just use the order of the headers in the
```

Comment:
Works for me, Let's clean up this comment with a clear description of how we expect this to function.